provider/lxd/credentials_test.go

```diff
@@ -79,8 +75,11 @@ func (s *credentialsSuite) TestDetectCredentialsUsesJujuCert(c *gc.C) {
 
 	deps := s.createProvider(ctrl)
 
-	deps.server.EXPECT().GetCertificate(s.clientCertFingerprint(c)).Return(nil, "", nil)
-	deps.server.EXPECT().ServerCertificate().Return("server-cert")
+	server := lxd.NewMockServer(ctrl)
```

Comment:
I would:
- Make the server a member of ```credentialSuiteDeps```.
- Instantiate it in ```s.createProvider```.
- Do ```serverFactory.EXPECT().LocalServer().Return(server, nil).AnyTimes()``` in ```s.createProvider``` too.

Should DRY up the tests a bit.

Reply:
agreed